### ISSUE TITLE
Namespace creation fails in e2e tests

### ISSUE THREAD

**Issue by satnam6502** [GoogleCloudPlatform/kubernetes#10289]
I have seen several e2e test failures which seem to be due to the namespace creation step in the test framework failing because the API server request fails.

@quinton-hoole @ixdy 

```
Cluster level logging using Elasticsearch 
  should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:39
[BeforeEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:47
STEP: Creating a kubernetes client
>>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-e2e-gce-flaky/workspace/.kube/config
STEP: Building a namespace api object
[AfterEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:48

• Failure in Spec Setup (BeforeEach) [0.153 seconds]
Cluster level logging using Elasticsearch
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:40
  should check that logs from pods on all nodes are ingested into Elasticsearch [BeforeEach]
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:39

  Expected error:
      <*url.Error | 0xc2084ff530>: {
          Op: "Post",
          URL: "https://104.197.0.135/api/v1/namespaces",
          Err: {
              Op: "dial",
              Net: "tcp",
              Addr: {
                  IP: "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xffh\xc5\x00\x87",
                  Port: 443,
                  Zone: "",
              },
              Err: 0x6f,
          },
      }
      Post https://104.197.0.135/api/v1/namespaces: dial tcp 104.197.0.135:443: connection refused
  not to have occurred

  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:63
```

**Comment by quinton-hoole** [comment]
Yes, I think that we need a retry loop around namespace creation and deletion.  And we need to factor that out into the Framework in test/utils.  I think the creation and deletion is done inside each test's BeforeEach()/AfterEach() right now.

**Comment by wojtek-t** [comment]
Fixed by #10301
